lib/compat/wordpress-6.5/blocks.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Block functions specific for the Gutenberg editor plugin.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers a new block style for one or more block types.
+ *
+ * @since 6.5.0
+ *
+ * @param string|array $block_name       Block type name including namespace or array of namespaced block type names.
+ * @param array        $style_properties Array containing the properties of the style name, label,
+ *                                 style_handle (name of the stylesheet to be enqueued),
+ *                                 inline_style (string containing the CSS to be added),
+ *                                 style_data (theme.json-like object to generate CSS from).
+ * @return bool True if all block styles were registered with success and false otherwise.
+ */
+function gutenberg_register_block_style( $block_name, $style_properties ) {
```

Comment:
Initially, this extended block style registration also generated a stylesheet and enqueued it for block style variations that included a style object within its properties. This allowed only a single set of styles to be generated for a block style variation that was shared across multiple block types.

Future follow-ups to this PR will be to allow block style variations to be defined within theme.json outside of a block type. These definitions can then be referenced across multiple block types.

Reply:
> Future follow-ups to this PR will be to allow block style variations to be defined within theme.json outside of a block type. These definitions can then be referenced across multiple block types.

That will be very handy! I came across that in testing with your provided test style variations and it was a bit confusing at first when tweaking styles in one of the Group variations had no effect in Columns with the same variation 😅

Reply:
> it was a bit confusing

Apologies for the confusion. I'll add a warning to the PR description for that gotcha.

I'm inclined to see if we can refine this to a point we're happy to land, while concurrently working on allowing the referenced variations within theme.json.

What do you think?